vignettes/teal-report-class.Rmd

```diff
@@ -0,0 +1,122 @@
+---
+title: "teal_report Class"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{teal_report Class}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  eval = FALSE
+)
+library(teal.reporter)
+```
+
+## Introduction
+
+The `teal_report` class in `teal.reporter` provides a way to create reproducible reports by adding, editing and removing document elements.
+
+The `teal_report` class is built on top of [`teal_data`](https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data.html), inheriting all its reproducibility and code-tracking capabilities while adding reporting-specific functionality through `teal_card`.
+
+This vignette shows you how to use `teal_report` and `teal_card` to enhance your teal modules with reporting capabilities.
+
+## Creating a teal_report
+
+Creating a `teal_report` is straightforward - you can include your data and it automatically tracks the code needed to reproduce your analysis:
+
+```{r basic-usage}
+library(teal.reporter)
+
+# Create a teal_report with data
+report <- teal_report(
+  iris = iris,
+  mtcars = mtcars
+)
+
+# You can also start with an empty report
+empty_report <- teal_report()
+```
+
+## Working with teal_card
+
+The `teal_card` is where your report content lives. You can create cards with text, code, results, and any other content:
+
+```{r teal-card-basics}
+# Create a simple card
+card <- teal_card(
+  "## My Analysis",
+  "This analysis looks at the iris dataset.",
+  summary(iris)
+)
+
+# Add more content to an existing card
+card <- c(card, list(
+  "### Conclusion", 
+  "The data shows interesting patterns."
+))
+
+# Add the card to a report
+report <- teal_report(iris = iris)
+teal_card(report) <- card
+```
+
+## Adding Computed Results with eval_code
+
+Since `teal_report` inherits from `teal_data`, you can use `eval_code()` to add computed results that are automatically tracked for reproducibility:
+
+```{r eval-code-example}
+library(teal.code)
+
+# Start with a teal_report
+report <- teal_report(iris = iris)
+
+# Add computed results using eval_code
+report <- eval_code(report, "
+  # Compute summary statistics
+  iris_summary <- summary(iris)
+  
+  # Create a simple model
+  model <- lm(Sepal.Length ~ Sepal.Width, data = iris)
+  model_summary <- summary(model)
+")
```

Comment:
Somebody could be mislead that `eval_code` can contain markdown syntax (comments which looks like heading in this context).
```suggestion
report <- eval_code(report, "
  iris_summary <- summary(iris)
  model <- lm(Sepal.Length ~ Sepal.Width, data = iris)
  model_summary <- summary(model)
")
```